### ISSUE TITLE
Samorost 3 (421120)

### ISSUE THREAD

**Issue by oblitum** [ValveSoftware/Proton#1888]
# Compatibility Report
- Name of the game with compatibility issues: Samorost 3
- Steam AppID of the game: 421120

## System Information
- GPU:  Intel HD Graphics 630 (i7-7700K)
- Driver/LLVM version: mesa 18.2.3
- Kernel version: 4.18.16
- Proton version: 3.16-4
- ArchLinux

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms

Game doesn't start, an application window reports that an Adobe AIR installation is required for the game.

## Solution

Assure `lcms2` and `lib32-lcms2` are installed in your system. The bundled Adobe AIR used by the game requires it.

## Former Solution

- Download [`winetricks` script](https://raw.githubusercontent.com/Winetricks/winetricks/master/src/winetricks), put it on your `PATH` and set execution permission for it
- Have `lcms2` and `lib32-lcms2` installed on your system (`pacman -S lcms2 lib32-lcms2` on ArchLinux)
- Add a new override for library `mscms` in the `winecfg` "Libraries" tab
  ```
  WINEPREFIX="/mnt/seagate/Steam/steamapps/compatdata/421120/pfx" \
  WINEARCH=win64 \
  "/mnt/seagate/Steam/steamapps/common/Proton 3.16/dist/bin/wine" winecfg
  ```
- Use the `winetricks` script to install Adobe AIR on the Samarost 3's WINE prefix
  ```
  WINE="/mnt/seagate/Steam/steamapps/common/Proton 3.16/dist/bin/wine" \
  WINEPREFIX="/mnt/seagate/Steam/steamapps/compatdata/421120/pfx" \
  WINEARCH=win64 \
  winetricks adobeair
  ```
After installing Adobe AIR successfully on Semarost 3's WINE prefix by using the WINE that comes with Proton, the game runs without issues.

**Comment by nsivov** [comment]
@oblitum do you know if manually installation of Air is necessary on Windows too, or should it happen automatically?

**Comment by oblitum** [comment]
@nsivov I have no idea...

**Comment by nsivov** [comment]
@oblitum I just tried on Windows, and it does not require manual installation of AIR. Then I tried with Proton 3.16-4 and it starts fine for me as is. I noticed you're using external/additional hdd for steam library, is it NTFS partition by any chance? In any case please try installing the game to the same partition with Steam.

**Comment by oblitum** [comment]
@nsivov Yes it's an external NTFS HDD, and the only one that can hold my games for now, I can't move them. Despite that, it has always been working fine (besides a glitch I had once, which [I've fixed applying my user/group as owner to the HDD](https://github.com/ValveSoftware/Proton/issues/35#issuecomment-414949271)). Do you think this instance of not automatically installing Adobe AIR is due to that? It's weird because I have a ton of games and the only one that doesn't auto-install such thing is this one. But as of now it's not a problem anymore since I've installed it manually without problems.

I have some other games that have had issues with missing .NET Framework, but I don't think HDD is related, [many people are reporting the same](https://github.com/ValveSoftware/Proton/issues/18).

**Comment by nsivov** [comment]
AIR runtime is bundled with the game, as far as I can tell it's a single library and not an installer. For testing purposes you can add another library folder on your Linux filesystem, and install only this game there. And yes, we got some reports with issues on NTFS specifically.

**Comment by oblitum** [comment]
Another thing is that at first I didn't have `lib32-lcms2` installed in my system, and I noticed that manual Adobe AIR install/use failed due to that in the winetricks messages. So `lcms2` et. al. was a required step. Can't say whether that would affect the game installation process since, when I installed Adobe AIR to the prefix, I notice the dependency, installed it and kept going with the manual procedure until completion.

**Comment by oblitum** [comment]
@nsivov Thanks for asking to try again. I followed my suspicion and just nuked the prefix and started the game from steam again. It started fine, so indeed, the fix is the missing `lib32-lcms2` that wasn't in my system at first. It's not related to NTFS either.
